src/interface/src/app/plan/scenario-results/scenario-results.component.ts

```diff
@@ -1,27 +1,69 @@
 import { Component, Input, OnChanges, SimpleChanges } from '@angular/core';
 import { ProjectAreaReport } from '../project-areas/project-areas.component';
-import { generateDummyData } from './temp-helper';
-import { ScenarioResult } from '../../types';
+import { Scenario, ScenarioResult } from '../../types';
+import { PlanService } from 'src/app/services';
+import { take } from 'rxjs';
 
 @Component({
   selector: 'app-scenario-results',
   templateUrl: './scenario-results.component.html',
   styleUrls: ['./scenario-results.component.scss'],
 })
 export class ScenarioResultsComponent implements OnChanges {
-  @Input() results: ScenarioResult | null = null;
+  @Input() scenario: Scenario | null = null;
 
   areas: ProjectAreaReport[] = [];
+  scenarioOutputFieldsConfigs: any = {};
 
-  // TODO remove and use real `this.results`
-  data = generateDummyData();
-  // start with the first 4 as selected
-  selectedCharts = this.data.slice(0, 4);
+  labels: string[][] = [];
+  data: any[] = [];
+  selectedCharts: any[] = [];
+
+  constructor(private planService: PlanService) {}
 
   ngOnChanges(changes: SimpleChanges) {
     // parse ScenarioResult
-    if (this.results) {
-      this.areas = this.parseResultsToProjectAreas(this.results);
+    if (this.scenario?.scenario_result) {
+      this.areas = this.parseResultsToProjectAreas(
+        this.scenario?.scenario_result
+      );
+      var scenario_output_fields_paths =
+        this.scenario.configuration.treatment_question
+          ?.scenario_output_fields_paths!;
+      this.planService
```

Comment:
I believe this should be done outside this component, and have scenario results be a dumb component that only takes inputs and deals with how to display them. Dumb components are way easier to test and manage, we can still have some formatting and processing to get the shapes that we need in here though.

Ideally, the service call and general shape should be done elsewhere, on the parent of this component.
Also, more importantly, by adding `getMetricData`  here on `ngChanges` you are doing a request each time anything under `this.scenario` changes which is not great

Reply:
Good call! I've moved it to the create-scenario component. Let me know if that looks better